packages/relay/src/lib/eth.ts

```diff
@@ -517,6 +518,10 @@ export class EthImpl implements Eth {
           return predefined.INVALID_PARAMETER(0, `Invalid 'value' field in transaction param. Value must be greater than 0`);
         }
       } else {
+        // execute reverted: with no message comes from a contract deployment
+        if (e.message === EthImpl.errorContractReverted) {
```

Comment:
> A 400 response can also include the following:

That's fine. If there were invalid params shouldn't we still return an error with those details? It's not a contract revert but it is a 400 error from the Mirror Node.
I believe we want to return the error with details for whatever reason that the Mirror Node returns except unsupported, which is a 501 case.
Is there a scenario still in the 400 case where we should not return the error?

In the current logic will all contract reverts be caught with this message check?
What if the mirror node changes the message text, this logic would break right?

> The e.isContractReverted() is used in the eth_call, via callMirrorNode, so that would have to be tested. eth_estimateGas simply calls the mirrorNode using the mirroNodeClient.

callMirrorNode calls `this.mirrorNodeClient.postContractCall`, which is the same call the estiamteGas is making via the  mirroNodeClient. It's the same mirror node endpoint in both cases